gridspeccer/core.py

```diff
@@ -129,87 +135,105 @@ def make_axes(gridspec, fig_kwargs=None):
-                fontdict=fontdict)
-
-def plot_caption(ax, caption, xpos=.04, ypos=.88, color="k", fontdict={}):
+                label_xpos=None, label_ypos=None, label_color=None, fontdict=None):
+    "plot labels"
+    label_xpos = label_xpos if label_xpos is not None else {}
+    label_ypos = label_ypos if label_ypos is not None else {}
+    label_color = label_color if label_color is not None else {}
+
+    for label_idx, char in zip(labels_to_plot,
+                               string.ascii_lowercase):
+        log.info("Subplot %s receives label %s", label_idx, char)
+        plot_caption(axes[label_idx], "\\textbf{" + char + "}",
+                     label_xpos.get(label_idx, xpos_default),
+                     label_ypos.get(label_idx, ypos_default), label_color.get(label_idx, "k"),
+                     fontdict=fontdict)
+
+
+def plot_caption(axis, caption, xpos=.04, ypos=.88, color="k", fontdict=None):
+    "plot caption"
     # find out how our caption will look in reality
-    caption_args={
-            "ha" : "left", "va" : "bottom",
-            #"weight" : "bold",
-            "style" : "normal",
-            "size" : 16,
-            "color": color,
-            "zorder" : 1000,
-        }
-    # r = get_renderer(ax.figure)
+    caption_args = {
+        "ha": "left",
+        "va": "bottom",
+        # "weight": "bold",
+        "style": "normal",
+        "size": 16,
+        "color": color,
+        "zorder": 1000,
+    }
+    # r = get_renderer(axis.figure)
     # bb = t.get_window_extent(renderer=r)
 
     # if fontdict is None:
-        # fontdict = {"family": "Linux Biolinum Kb"}
-        # size = caption_args["size"]
-        # bbox = mpatches.FancyBboxPatch(ax.transAxes.transform((xpos, ypos)),
-                # size *1.0, size*1.0, zorder=10,
-                # edgecolor="k", facecolor="r", boxstyle="round")
-        # ax.patches.append(bbox)
+    #     fontdict = {"family": "Linux Biolinum Kb"}
+    #     size = caption_args["size"]
+    #     bbox = mpatches.FancyBboxPatch(axis.transAxes.transform((xpos, ypos)),
+    #             size *1.0, size*1.0, zorder=10,
+    #             edgecolor="k", facecolor="r", boxstyle="round")
+    #     axis.patches.append(bbox)
+
+    axis.text(xpos, ypos, caption, fontdict=fontdict, transform=axis.transAxes,
+              **caption_args)
+
+
+def hide_axis(axis):
+    "hide axis"
+    axis.get_xaxis().set_visible(False)
+    axis.get_yaxis().set_visible(False)
+
 
-    t = ax.text(xpos, ypos, caption, fontdict=fontdict, transform=ax.transAxes,
-            **caption_args )
+def show_axis(axis):
+    "show axis"
+    axis.get_xaxis().set_visible(True)
+    axis.get_yaxis().set_visible(True)
 
-def hide_axis(ax):
-    ax.get_xaxis().set_visible(False)
-    ax.get_yaxis().set_visible(False)
 
-def show_axis(ax):
-    ax.get_xaxis().set_visible(True)
-    ax.get_yaxis().set_visible(True)
+def hide_ticks(axis, axes='both', minormajor='both'):
+    "hide ticks"
```

Comment:
Very helpful docstring :stuck_out_tongue_closed_eyes:

Reply:
feel free to be more prosaic :)